soroban-env-host/src/events/system_events.rs

```diff
@@ -0,0 +1,45 @@
+use soroban_env_common::{xdr::ScContractExecutable, EnvBase, Symbol, TryFromVal, TryIntoVal};
+use soroban_native_sdk_macros::contracttype;
+
+use crate::{native_contract::base_types::BytesN, Host, HostError};
+
+const CONTRACT_EXECUTABLE_UPDATE_TOPIC: &str = "executable_update";
+
+#[contracttype]
+enum ContractExecutable {
+    WasmRef(BytesN<32>),
+    Token,
+}
+
+impl ContractExecutable {
+    fn from_xdr(host: &Host, xdr: &ScContractExecutable) -> Result<Self, HostError> {
+        match xdr {
+            ScContractExecutable::WasmRef(wasm_hash) => Ok(ContractExecutable::WasmRef(
+                BytesN::<32>::from_slice(host, &wasm_hash.0)?,
```

Comment:
This needs metering.

Reply:
Does it? I thought adding a host object is metered and otherwise that's just a trivial move.

Reply:
This code itself is okay. The adding host object part is metered. Although I found the `to_vec()` is doing a deep clone on the slice to create a new vector and should've been metered. I've commented on https://github.com/stellar/rs-soroban-env/issues/373#issuecomment-1507137743 to track it (there might be other places too). You don't have to do anything here.

Reply:
Sounds good, resolving this for now then - if there are some other budget metering gaps we haven't noticed here, let's address them as a followup.